core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushDownDereferencesThroughLimit.java

```diff
@@ -71,7 +72,9 @@ public PushDownDereferencesThroughLimit(TypeAnalyzer typeAnalyzer)
     public Pattern<ProjectNode> getPattern()
     {
         return project()
-                .with(source().matching(limit().capturedAs(CHILD)));
+                .with(source().matching(limit()
+                        .with(isOrderSensitive().equalTo(false))
```

Comment:
This should be allowed. The ordering scheme cannot depend on the dereferenced expressions, and the project won't change the order of the input rows.